src/common/conditional_compilation/docs/develop_cc_for_new_component.md

```diff
@@ -0,0 +1,204 @@
+They are very similar except for different region suffix, one is run on function, another is run on model. Both them leverage OV_SCOPE with ov_pass as its module name.
+
+https://github.com/openvinotoolkit/openvino/blob/34b76584f724ad07ed27c8ea27777ac46df92c23/src/common/conditional_compilation/include/openvino/cc/pass/itt.hpp#L20-L22
+
+
+### 2. Macro in SELECTIVE_BUILD mode
+
+#### OV_CC_DOMAINS
+It is empty in this mode.
+
+#### OV_CC_SCOPE_IS_ENABLED
+It checks whether the condition is true or false, it always is embedded in **OV_SCOPE** or **MATCHER_SCOPE** to decide whether the following code region need be excluded out of final binaries.
+
+#### OV_SCOPE
+It will check whether the code region in this module is active or inactive by the macros of header file generated in previous analyzed mode. The generated macros name format is <module>_<region>, in which module name is passed by the parameters of OV_SCOPE. 
+
+https://github.com/openvinotoolkit/openvino/blob/34b76584f724ad07ed27c8ea27777ac46df92c23/src/common/conditional_compilation/include/openvino/cc/selective_build.h#L181-L183
+
+
+#### OV_SWITCH
+It will decide whether the template class parameter match operation need be excluded from final binaries. If the match operation is false in analyzed mode, the code region will be excluded in SELECTIVE_BUILD mode.
+
+#### OV_CASE and OV_CASE2
+They are used with OV_SWITCH to provide match cases.
+
+#### MATCHER_SCOPE
+It defines a string to represent the matcher name in this region, and then decide whether the code region is inactivated and excluded from final binaries, according to the OV_SCOPE result in analyzed mode.
+
+https://github.com/openvinotoolkit/openvino/blob/34b76584f724ad07ed27c8ea27777ac46df92c23/src/common/conditional_compilation/include/openvino/cc/pass/itt.hpp#L30-L33
+
+
+#### RUN_ON_FUNCTION_SCOPE and RUN_ON_MODEL_SCOPE
+They are very similar, and functionality is same with MATCHER_SCOPE plus throw error.
+
+### 3. Non Conditional Compilation
+It means that conditional compilation is disabled.
+
+#### OV_CC_DOMAINS
+It is empty in this mode
+
+#### OV_SCOPE
+It is empty in this mode
+
+#### OV_SCOPE and OV_CASE
+They will do match operation without itt scope task.
+
+#### MATCHER_SCOPE
+It is empty.
+
+#### RUN_ON_FUNCTION_SCOPE and RUN_ON_MODEL_SCOPE
+They are empty.
+
+
+## Enable conditional complication for a new component or domain
+### step 1: requirement analyze
+Analyze which code region need conditional complication and figure out requirement for conditional complication.
+
+### step 2: create itt.hpp if possible
+Create itt.hpp file in the target component directory, in which you need define some particular macro needed by conditional compilation if existed macro cannot meet requirement. An example [itt.hpp](https://github.com/openvinotoolkit/openvino/blob/master/src/common/transformations/src/itt.hpp).
+
+- Add macro into itt.hpp to define new CC domain:
+```
+    #include <openvino/cc/selective_build.h>
```

Comment:
I think we can provide real code snippets for example from src/core/src/itt.hpp

Reply:
done